src/t8_forest/t8_forest_adapt.cxx

```diff
@@ -67,8 +67,12 @@ t8_forest_adapt_coarsen_recursive (t8_forest_t forest, t8_locidx_t ltreeid,
   element = t8_element_array_index_locidx (telements, *el_inserted - 1);
   /* TODO: This assumes that the number of children is the same for each
    *       element in that class. This may not be the case. */
-  num_children = ts->t8_element_num_children (element);
-  T8_ASSERT (ts->t8_element_child_id (element) == num_children - 1);
+  ts->t8_element_new(1, &parent);
+  T8_ASSERT(ts->t8_element_level(element)> 0);
+  ts->t8_element_parent(element, parent);
+  num_children = ts->t8_element_num_children (parent);
+  //T8_ASSERT (ts->t8_element_child_id (element) == num_children - 1);
+  ts->t8_element_destroy(1, &parent);
 
   fam = el_buffer;
```

Comment:
we should grow el_buffer when the number of siblings increases.
See also my other comments on max_num_siblings

Reply:
This Problem has been adressed in the commit https://github.com/holke/t8code/pull/89/commits/0015db4aa6354d801886ed079bfa0f4fea54588a